SIPS/sip-16.md

```diff
@@ -0,0 +1,131 @@
+These insights can then be displayed in the MetaMask UI alongside any information provided by MetaMask itself.
+Signature insight Snaps can also return a `severity` key to indicate the severity level of the content being returned. 
+This key uses the same `SeverityLevel` enum used by transaction insight Snaps. 
+
+## Specification
+
+> Formal specifications are written in Typescript. Usage of `CAIP-N` specifications, where `N` is a number, are references to [Chain Agnostic Improvement Proposals](https://github.com/ChainAgnostic/CAIPs).
+
+### Language
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT",
+"SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and
+"OPTIONAL" written in uppercase in this document are to be interpreted as described in [RFC 2119](https://www.ietf.org/rfc/rfc2119.txt)
+
+### Definitions
+
+> This section is non-normative, and merely recapitulates some definitions from [CAIP-2](https://github.com/ChainAgnostic/CAIPs/blob/master/CAIPs/caip-2.md).
+
+- `ChainId` - a [CAIP-2](https://github.com/ChainAgnostic/CAIPs/blob/master/CAIPs/caip-2.md) string.
+  It identifies a specific chain among all blockchains recognized by the CAIP standards. The ChainId refers to the network that the signing account is connected to. It is not necessarily the network where the signed message will be broadcast. 
+  - `ChainId` consists of a `Namespace` and a `Reference`
+    - `Namespace` - a class of similar blockchains. For example EVM-based blockchains.
+    - `Reference` - a way to identify a concrete chain inside a `Namespace`. For example Ethereum Mainnet or one of its test networks.
+- `Method` - a string which can be any of the following 5 [signing methods](https://docs.metamask.io/wallet/concepts/signing-methods/) supported by the MetaMask wallet API. The signing method is required by MetaMask for the dapp to make a signature request. It is not part of the signature payload.
+  1. `eth_signTypedData_v4`
+  2. `eth_signTypedData_v3`
+  3. `eth_signTypedData_v1`
+  4. `personal_sign`
+  5. `eth_sign`
+
+### Snap Manifest
+
+This SIP specifies a permission named `endowment:signature-insight`.
+The permission grants a Snap read-only access to raw signature payloads, before they are accepted for signing by the user.
+
+The permission is specified as follows in `snap.manifest.json` files:
+
+```json
+{
+  "initialPermissions": {
+    "endowment:signature-insight": {}
+  }
+}
+```
+
+The permission includes an OPTIONAL caveat `allowSignatureOrigin`. 
+The caveat grants a Snap read-only access to the URL requesting the signature.
+It can be specified as follows: 
+
+```json
+{
+  "initialPermissions": {
+    "endowment:signature-insight": {
+      "allowSignatureOrigin": true
+    }
+  },
+}
+```
+
+### Snap Implementation
+
+The following is an example implementation of the API:
+
+```typescript
+import { OnSignatureHandler } from "@metamask/snap-types";
+
+enum SeverityLevel {
```

Comment:
Import from `snaps-types` instead seems clearer

Reply:
Why was this resolved? Was there further conversation about this?

Reply:
I think you might have misunderstood what Frederik meant, I believe he was referring to importing the enum. It was then the right package, but now we use the sdk so no longer valid lol. Nonetheless, I've fixed in [6da2614](https://github.com/MetaMask/SIPs/pull/117/commits/6da261452194c304f43c28a220dfa55e26cf17ee)